### PR TITLE
Fix enterprise project activation

### PR DESCRIPTION
**Pull request by MDrakos**
https://www.pivotaltracker.com/story/show/173948845

### CODE REVIEW

File: pkg/platform/model/recipe.go [hunk 1]
```diff
@@ -133,17 +138,20 @@ func commitToOrder(commitID strfmt.UUID, owner, project string, hostPlatform *st
...
+func fetchRecipeID(commitID strfmt.UUID, owner, project, orgID string, private bool, hostPlatform *string) (*strfmt.UUID, *failures.Failure) {
 	var err error
 	params := iop.NewSolveOrderParams()
 	params.Order, err = commitToOrder(commitID, owner, project, hostPlatform)
 	if err != nil {
 		return nil, FailOrderRecipes.Wrap(err)
 	}
+	if private {
+		params.OrganizationID = &orgID
+	}
```
Comment (Naatan) [inline pkg/platform/model/recipe.go:150]: Any reason why we shouldn't just always set this?

File: pkg/platform/model/recipe.go [hunk 2]
```diff
@@ -133,17 +138,20 @@ func commitToOrder(commitID strfmt.UUID, owner, project string, hostPlatform *st
 	return order, nil
 }
 
-func fetchRecipeID(commitID strfmt.UUID, owner, project string, hostPlatform *string) (*strfmt.UUID, *failures.Failure) {
+func fetchRecipeID(commitID strfmt.UUID, owner, project, orgID string, private bool, hostPlatform *string) (*strfmt.UUID, *failures.Failure) {
 	var err error
 	params := iop.NewSolveOrderParams()
 	params.Order, err = commitToOrder(commitID, owner, project, hostPlatform)
```
Comment (Naatan) [inline pkg/platform/model/recipe.go:144]: We're still passing hostPlatform here, given the new additions for asserting the platform should this be removed?

File: pkg/platform/model/recipe.go [hunk 3]
```diff
@@ -165,15 +173,21 @@ func fetchRecipeID(commitID strfmt.UUID, owner, project string, hostPlatform *st
 		}
 	}
 
-	// Because we filter platforms in the request we should only
-	// receive one recipe ID
-	if len(recipeID.Payload) != 1 {
-		return nil, FailOrderRecipes.New("err_recipe_payload")
+	if hostPlatform == nil {
+		hostPlatform = &HostPlatform
```
Comment (Naatan) [inline pkg/platform/model/recipe.go:177]: I'd rather we rely on this always being passed in, so we're not making this function rely on globals.

File: pkg/platform/model/recipe.go (hunk 1 above)
Comment (MDrakos) [reply pkg/platform/model/recipe.go:150]: If this is set on public projects we do not get the correct recipeID back

File: pkg/platform/model/recipe.go (hunk 2 above)
Comment (MDrakos) [reply pkg/platform/model/recipe.go]: Originally `commitToOrder` was also filtering platforms from the returned order. It's no longer needed here, but is still needed in `fetchRawRecipes`, where it now lives until we address this: https://www.pivotaltracker.com/story/show/174076298.

**Pull request merged**